### ISSUE TITLE
Invalid mesh topology error

### ISSUE THREAD

**Issue by mhmthudayi** [mfem/mfem#4119]
Hello,

I am new in mfem and glvis libraries. I could visualize some of my simple data. However, when I try to visualize a little bit complex mesh and result in glvis, I get the following error:

Invalid mesh topology. Interior edge found between 2D elements 40, 50 and 100.

What exactly is the problem here, and how can I solve it? The geometry is like a T-beam which involves two plates attached to each other perpendicularly. I have created the files externally based on mfem mesh formats, they are not generated by mfem.

Thanks in advance for your help,
Best regards

**Comment by tzanio** [comment]
Hi @mhmthudayi,

Do you have a 2D mesh embedded in 3D?

This message says that 3 elements share an edge, which is not allowed for planar 2D meshes.

Tzanio

**Comment by mhmthudayi** [comment]
![image_2024-02-08_134628727](https://github.com/mfem/mfem/assets/159302254/02312577-f1d5-4866-bdaa-441680bdf11c)

My elements are 2D squares but the geometry is 3D as you can see in the screenshot. These 3 square elements must share that edge. I could visualize this by adding two more nodes to that connection line. However, this does not solve my problem. I need to define this connection without adding more nodes somehow. What should I use to overcome this challenge?

Thanks

**Comment by tzanio** [comment]
We don't currently support configurations like this in MFEM, but maybe we can, to some extend, with small changes.

What discretization do you plan to use on this mesh? Elements like H1 make sense, but using DG/L2 is less clear.

Tzanio

**Comment by mhmthudayi** [comment]
![result](https://github.com/mfem/mfem/assets/159302254/579fe92a-ec68-4753-aebd-3df36d95bbf8)

Ok, we have solved the issue. This is the mesh and result we tried to visualize using glvis. We manipulated the mesh with our in-house fem codes, and glvis successfully visualized it. 

Thanks

**Comment by tzanio** [comment]
> Ok, we have solved the issue. 

Great! Can you elaborate what you did and what the discretization is?

**Comment by mhmthudayi** [comment]
We first determined if an edge is shared by more than 2 elements. If it is, we created two more nodes at the same locations and constructed the third element using the nodes we added. Also, we assigned the displacement value of the added nodes according to the displacements of the actual nodes we have there. We don't use mfem to solve displacements on this mesh, we compute them somewhere else using the original mesh, then manipulate the mesh and assign results to the new nodes only for visualization purposes on glvis. Discretization is done with quadrilateral elements configured in 3D space, so they have 3D displacements etc.
